### DISCUSSION TITLE
use template from GitHub stopped working for me

### DISCUSSION THREAD

**Discussion by rubinjm** [quarto-dev/quarto-cli#10043] [Q&A] [answered]
### Description

I have a GitHub repository that contains a quarto template and had been entering the 'use template' command in the R Studio terminal to install it without any problems. Now I'm getting an error message and not sure why or how to fix it. Not sure if it started after an RStudio upgrade? (I think I originally entered a command to make it work but can't find those instructions.)

`[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.13: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.32
      Path: /Applications/quarto/bin`

**Error when I run the command:**
`$ quarto use template XXXXX/quarto-template

Quarto templates may execute code when documents are rendered. If you do not 
trust the authors of the template, we recommend that you do not install or 
use the template.
? Do you trust the authors of this template (Y/n) › Yes
? Create a subdirectory for template? (Y/n) › No
[✓] Downloading
[✓] Unzipping
ERROR: NotFound: No such file or directory (os error 2): readdir '/var/folders/b7/h6ynt7bs0zlctj73p_xst0dm0000gp/T/quarto-sessiona7469891fe279f92/fa347fff22e2855c/4489e2675f22d46c/archive/quarto-template-main/_extensions'

Stack trace:
    at Object.readDirSync (ext:deno_fs/30_fs.js:211:10)
    at readExtensions (file:///Applications/quarto/bin/quarto.js:42853:32)
    at useTemplate (file:///Applications/quarto/bin/quarto.js:109514:42)
    at eventLoopTick (ext:core/01_core.js:153:7)
    at async Command.actionHandler (file:///Applications/quarto/bin/quarto.js:109497:9)
    at async Command.execute (file:///Applications/quarto/bin/quarto.js:8015:13)
    at async Command.parseCommand (file:///Applications/quarto/bin/quarto.js:7905:20)
    at async quarto (file:///Applications/quarto/bin/quarto.js:117893:5)
    at async file:///Applications/quarto/bin/quarto.js:117905:9
    at async mainRunner (file:///Applications/quarto/bin/quarto.js:117798:9)`

**Comment by cderv** [comment, marked as answer]
Please update your Quarto version, especially when you are using a pre-release version. You are suing 1.5.32 and we are now at 1.5.45

This has been fixed already in one of the version in-between. 
- https://github.com/quarto-dev/quarto-cli/issues/9256

**Reply by rubinjm** [reply]
Oh thanks--I thought Quarto automatically updated to the latest version along with RStudio updates?

**Reply by mcanouil** [reply]
The two are not synced and RStudio does not embeds Pre-release versions.

You are using pre-release, so you manually opt-in to use that version, thus needs to update it regularly.

**Reply by cderv** [reply]
RStudio IDE bundles stable, but it depends on timing on RStudio IDE updates regarding Quarto updates. They may not include the latest stable release, or there could be delay between both. 

Quarto is bundled by convenience, but you will be well served if you managed your Quarto install on your own.

**Comment by cdriveraus** [comment]
For anyone else who ends up here, you need the development version of the quarto R package:

```
remotes::install_github("quarto-dev/quarto-r")
```

**Reply by mcanouil** [reply]
Sorry, but the R package has nothing to do here.

To be clear, this is a version issue with Quarto CLI.

**Comment by cdriveraus** [comment]
I had the same error. The most recent CRAN quarto release uses 1.4.4. Installing the dev version fixed the error. am I going mad?

**Reply by cderv** [reply]
You're not going crazy - but you are not encountering the same issue as described initially. You may encounter this problem 
- https://github.com/quarto-dev/quarto-r/issues/212

which has been fixed in dev version of Quarto R package. 

This is a problem with extension approval. That is the only related fix I see in dev version. 

@cdriveraus same for you. You may need the R package dev version if you had this issue. But you are not supposed to get `ERROR: NotFound: No such file or directory` initial issue anymore. 

Hope it helps

**Reply by mcanouil** [reply]
To be clear and for future readers, the issue reported in this thread is/was about Quarto CLI.

The second error reported, as pointed out by Christophe, is related to the `quarto` R package calling Quarto CLI, not Quarto CLI itself.

The two things are independent issues.
